src/main/java/org/commcare/xml/CaseXmlParser.java

```diff
@@ -209,6 +211,27 @@ private void updateCase(Case caseForBlock, String caseId) throws InvalidStructur
                     break;
             }
         }
+        checkForMaxLength(caseForBlock);
+    }
+
+    private void checkForMaxLength(Case caseForBlock) throws InvalidStructureException {
+        if (caseForBlock.getTypeId().length() > 255) {
+            throw new InvalidStructureException(
+                    "case_type is longer than 255 chars for case with ID: " + caseForBlock.getCaseId() + " and case_type " + caseForBlock.getTypeId(),
```

Comment:
I'd rephrase these based on the action they encode

IE: "Invalid <case_type>, value must be 256 characters or less"